### ISSUE TITLE
PR from a forked repository of an internal or a private repo is not getting detected on Jenkins using Github App

### ISSUE THREAD

**Issue by pchugh17** [jenkinsci/scm-filter-aged-refs-plugin#35]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
  Paste the output here
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

OS: RHEL9

### Reproduction steps

1. Create a repository with visibility set as internal or private
2. Fork this to a user account
3. Make changes & create a PR to the parent repo
4. Jenkins Org Job doesn't auto detect the new PR. If we do a repo scan or an org scan, it fails with below stacktrace:
 ```
Error while processing pull request 11
java.io.FileNotFoundException: https://<git_domain>/api/v3/repos/<user-account>/<repo_name>/commits/3f02475aad8f10af7b57c3fd4dfd57bcd131d190
	at org.kohsuke.github.GitHubConnectorResponseErrorHandler$1.onError(GitHubConnectorResponseErrorHandler.java:60)
	at org.kohsuke.github.GitHubClient.detectKnownErrors(GitHubClient.java:504)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:464)
Caused: org.kohsuke.github.GHFileNotFoundException: https://<git_domain>/api/v3/repos/<user-account>/<repo_name>/commits/3f02475aad8f10af7b57c3fd4dfd57bcd131d190 {"message":"Not Found","documentation_url":"https://docs.github.com/enterprise-server@3.12/rest/commits/commits#get-a-commit"}
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:661)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:480)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:427)
	at org.kohsuke.github.Requester.fetch(Requester.java:85)
	at org.kohsuke.github.GHRepository.getCommit(GHRepository.java:2101)
	at org.kohsuke.github.GHCommitPointer.getCommit(GHCommitPointer.java:100)
	at org.jenkinsci.plugins.scm_filter.GitHubAgedRefsTrait$ExcludeOldBranchesSCMHeadFilter.isExcluded(GitHubAgedRefsTrait.java:88)
	at jenkins.scm.api.trait.SCMSourceRequest.isExcluded(SCMSourceRequest.java:196)
	at jenkins.scm.api.trait.SCMSourceRequest.process(SCMSourceRequest.java:333)
	at org.jenkinsci.plugins.github_branch_source.GitHubSCMSource.retrievePullRequest(GitHubSCMSource.java:1245)
	at org.jenkinsci.plugins.github_branch_source.GitHubSCMSource.retrieve(GitHubSCMSource.java:1085)
	at jenkins.scm.api.SCMSource._retrieve(SCMSource.java:372)
	at jenkins.scm.api.SCMSource.fetch(SCMSource.java:282)
	at jenkins.branch.MultiBranchProject.computeChildren(MultiBranchProject.java:654)
	at com.cloudbees.hudson.plugins.folder.computed.ComputedFolder.updateChildren(ComputedFolder.java:269)
	at com.cloudbees.hudson.plugins.folder.computed.FolderComputation.run(FolderComputation.java:167)
	at jenkins.branch.MultiBranchProject$BranchIndexing.run(MultiBranchProject.java:1057)
	at hudson.model.ResourceController.execute(ResourceController.java:101)
	at hudson.model.Executor.run(Executor.java:443)
```
5. remove the plugin config (Filter by ref age), and scan again, PR is shown on Jenkins.

### Expected Results

Every new PR should be available on Jenkins

### Actual Results

Repo scan is failing & PR is not showing up on Jenkins for CI build

**Comment by pchugh17** [comment]
Hi @TobiX , do you got a chance to check this? Can we not catch the exceptions and return false for such cases, it will make sure PRs don't get skipped in such cases.

```
Iterable<GHPullRequest> pulls = ((GitHubSCMSourceRequest) scmSourceRequest).getPullRequests();
for (GHPullRequest pull : pulls) {
    if (("PR-" + pull.getNumber()).equals(scmHead.getName())) {
        try {
            long pullTS = pull.getHead()
                              .getCommit()
                              .getCommitShortInfo()
                              .getCommitDate()
                              .getTime();
            return pullTS < super.getAcceptableDateTimeThreshold();
        } catch (FileNotFoundException e) {
            // Handle the exception - log it, rethrow it, or handle it as needed
            System.err.println("Commit not found for pull request #" + pull.getNumber() + ": " + e.getMessage());
            e.printStackTrace();
        } catch (IOException e) {
            // Handle other IOExceptions
            System.err.println("IO exception occurred for pull request #" + pull.getNumber() + ": " + e.getMessage());
            e.printStackTrace();
        }
    }
}

// Handle the case where no matching pull request is found or other fallback logic
return false;
```

Additionally, I was scanning the code of github-branch-source-plugin, and getCommit is called on GHRepo object instead, will that help? Check [this](https://github.com/jenkinsci/github-branch-source-plugin/blob/5a7859812c8d1365e0a578787e509f10ae81b6d0/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMProbe.java#L105-L131) code for ex.

**Comment by TobiX** [comment]
@pchugh17 If you are still interested in a fix, could you try a build from MR #43 (https://ci.jenkins.io/job/Plugins/job/scm-filter-aged-refs-plugin/job/private-repo-pr/)?
